wasm/tx_ibc/src/lib.rs

```diff
@@ -3,51 +3,59 @@
 //! tx_data. This tx uses an IBC message wrapped inside
 //! `key::ed25519::SignedTxData` as its input as declared in `ibc` crate.
 
+use std::collections::BTreeMap;
+
 use namada_tx_prelude::action::{Action, MaspAction, Write};
-use namada_tx_prelude::token::UnshieldingTransferData;
 use namada_tx_prelude::*;
 
 #[transaction]
 fn apply_tx(ctx: &mut Ctx, tx_data: BatchedTx) -> TxResult {
     let data = ctx.get_tx_data(&tx_data)?;
-    let (transfer, masp_fee_payment) =
+    let transfer =
         ibc::ibc_actions(ctx).execute(&data).into_storage_result()?;
 
-    if let Some(UnshieldingTransferData {
-        token,
-        amount,
-        target,
-    }) = &masp_fee_payment
-    {
-        // Transparent unshield for fee payment
-        token::transfer(
-            ctx,
-            &Address::Internal(address::InternalAddress::Masp),
-            target,
-            token,
-            amount.amount(),
-        )?;
-    }
+    if let Some(transfers) = transfer {
+        // Prepare the sources of the multi-transfer
+        let sources = transfers
+            .sources
+            .into_iter()
+            .map(|(account, amount)| {
+                ((account.owner, account.token), amount.amount())
+            })
+            .collect::<BTreeMap<_, _>>();
+
+        // Prepare the target of the multi-transfer
+        let targets = transfers
+            .targets
+            .into_iter()
+            .map(|(account, amount)| {
+                ((account.owner, account.token), amount.amount())
+            })
+            .collect::<BTreeMap<_, _>>();
+
+        // Effect the multi transfer
+        token::multi_transfer(ctx, &sources, &targets)
+            .wrap_err("Token transfer failed")?;
```

Comment:
This change wasn't included in #3459 or #3463. unexpected change?

Reply:
You're right. I made this change when I was trying to merge #3459 and #3463 with the MASP fee payment changes (that were also touching IBC code). This code is basically an extension of what's already in `draft`.

Reply:
ah, I see. The transfer has been added for the fee payment in #3393. However, `masp_fee_payment` was discarded.
```rust
    if let Some(UnshieldingTransferData {
        token,
        amount,
        target,
    }) = &masp_fee_payment
    {
        // Transparent unshield for fee payment
        token::transfer(
            ctx,
            &Address::Internal(address::InternalAddress::Masp),
            target,
            token,
            amount.amount(),
        )?;
    }
```

Reply:
Yep, exactly. @grarco Let me know if you have any objections to this merge resolution. Essentially it removes the redundant `Transfer` (that isn't executed by `tx_ibc`) from `MsgTransfer` (and similar IBC messages).

Reply:
In general it looks ok to me, since the new `Transfer` object is vectorized we can embed the fee payment directly inside it. But the `run_ledger_ibc_with_hermes` test is failing and it contains a masp fee payment over ibc so it's better if we check it